### ISSUE TITLE
pillow-heif library install error

### ISSUE THREAD

**Issue by zigzag414** [malcolmosh/dispatchPi#5]
Getting this error when trying to install the pillow-heif library:

` sudo apt python3 -m pip install --upgrade pillow-heif --no-binary :all:`

`E: Command line option --no-binary is not understood in combination with the other options`

Coming up empty with my troubleshooting. Any recommendations?

**Comment by malcolmosh** [comment]
Try running the command without the "apt" : 

`sudo python3 -m pip install --upgrade pillow-heif --no-binary :all:
`

**Comment by zigzag414** [comment]
Gave that a try and the install errored out. I checked the update status for `wheels` too and got the same output. 
![er1](https://github.com/malcolmosh/dispatchPi/assets/94332953/d81b7d89-b31f-4a09-af19-a66089333507)
![er2](https://github.com/malcolmosh/dispatchPi/assets/94332953/4522d0b8-db91-4f17-bd2a-6bf9130a2ce2)

**Comment by malcolmosh** [comment]
Hmm. Could you paste the error log in text ?

Some potential solutions :

1. Run these commands first (remove an extraneous "install" and also added python3-dev)

```
sudo apt-get update && sudo apt upgrade -y
sudo apt install python3-pip python3-venv
sudo apt-get install python3-pil python3-numpy libffi-dev libheif-dev libde265-dev python3-venv libjpeg-dev zlib1g-dev git-all
sudo apt-get install python3-dev
```

2. Try installing without the no-binary flag : 

`sudo python3 -m pip install --upgrade pillow-heif`

3.  Try installing a specific library version : 

`sudo python3 -m pip install pillow-heif==0.11.1`

4. Try intalling the latest version directly:

`sudo python3 -m pip install pillow-heif`

5. Check the installed version of libheif on the Raspberry Pi and compare it to the latest requirements for pillow-heif (in their github repo).

`dpkg -l | grep libheif`

**Comment by malcolmosh** [comment]
I just checked on my frame and here are the library versions installed:

-  libheif1 : 1.11.0.1
- python3-pip: 20.3.4-4+rpt1+deb11u1
- python3-venv: 3.9.2-3
- python3-pil: 8.1.2+dfsg-0.3+deb11u1
- python3-numpy: 1:1.19.5-1
- libffi-dev: 3.3-6
- libheif-dev: 1.11.0-1
- libde265-dev: 1.0.11-0+deb11u1+rpi1
- libjpeg-dev: 1:2.0.6-4
- zlib1g-dev: 1:1.2.11.dfsg-2+deb11u2
- libpython3-dev: 3.9.2-3
- python3-dev: 3.9.2-3

The command below would help reproduce my pi's environment. Sorry I don't have a docker image. 

```
sudo apt-get update
sudo apt-get install \
python3-pip=20.3.4-4+rpt1+deb11u1 \
python3-venv=3.9.2-3 \
python3-pil=8.1.2+dfsg-0.3+deb11u1 \
python3-numpy=1:1.19.5-1 \
libffi-dev=3.3-6 \
libheif-dev=1.11.0-1 \
libde265-dev=1.0.11-0+deb11u1+rpi1 \
libjpeg-dev=1:2.0.6-4 \
zlib1g-dev=1:1.2.11.dfsg-2+deb11u2 \
libpython3-dev=3.9.2-3 \
python3-dev=3.9.2-3
```

**Comment by malcolmosh** [comment]
Hey @zigzag414 , did you get around to trying these fixes?

**Comment by zigzag414** [comment]
yeah I rolled everything to those versions with no change to the output unfortunately. i'm going to re-image the pi and go through the steps again carefully to see if I get the same result. Will report back.

**Comment by malcolmosh** [comment]
Ok good! I'll try to run it myself. You could always drop pillow-heif as a last resort, it's meant for compatibility with HEIC images.

**Comment by malcolmosh** [comment]
Here are two related threads as well:

- https://github.com/bigcat88/pillow_heif/issues/27
- https://github.com/bigcat88/pillow_heif/issues/147

You could try installing via the wheels linked in the first thread?

**Comment by malcolmosh** [comment]
Hi again. I spent a few hours re-installing everything on a Pi Zero W, and it turns out the processor may simply be incompatible with the pillow-heif library for the moment. I haven't managed to install it properly despite trying all kinds of strategies. I've updated the tutorial to suggest skipping this library locally and just making sure you only use jpg or png files on your local Pi, instead of HEIC/HEIF images. In the fully deployed app, local images are only used as backup anyway. Sorry about that!
